stellargraph/layer/deep_graph_infomax.py

```diff
@@ -0,0 +1,156 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from . import GCN, GAT, APPNP, PPNP
+from ..core.experimental import experimental
+
+from tensorflow.keras.layers import Input, Lambda, Layer, GlobalAveragePooling1D
+import tensorflow as tf
+from tensorflow.keras import backend as K
+import warnings
+
+__all__ = ["DeepGraphInfoMax", "BilinearTransform"]
+
+
+class BilinearTransform(Layer):
+    """
+    This Layer computes the Discriminator function for Deep Graph Infomax (https://arxiv.org/pdf/1809.10341.pdf).
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def build(self, input_shapes):
+
+        self.kernel = self.add_weight(
+            shape=(input_shapes[0][-1], input_shapes[0][-1]),
+            initializer="glorot_uniform",
+            name="kernel",
+            regularizer=None,
+            constraint=None,
+        )
+        self.built = True
+
+    def call(self, inputs):
+        """
+        Applies the layer to the inputs.
+
+        Args:
+            inputs: a list or tuple of tensors with shapes [(N, F), (F,)] containing the node features and summary feature
```

Comment:
I think these shapes are now slightly wrong?

```suggestion
            inputs: a list or tuple of tensors with shapes [(batch, N, F), (batch, F,)] containing the node features and summary feature
```

maybe?